server/handlerequest.go

```diff
@@ -34,41 +35,41 @@ func handleRequest(r *repo.Repo, handler Handler, jsonBytes []byte, source strin
 	// case HandlerGetRepo: // This case is handled prior to handleRequest being called.
 	// since the resulting bytes are written directly in to the http.ResponseWriter / net.Connection
 	case HandlerGetURIs:
-		getURIRequest := &requests.URIs{}
-		processIfJSONIsOk(json.Unmarshal(jsonBytes, &getURIRequest), func() {
+		getURIRequest := new(requests.URIs)
+		processIfJSONIsOk(json.NewDecoder(rdr).Decode(getURIRequest), func() {
 			reply = r.GetURIs(getURIRequest.Dimension, getURIRequest.IDs)
 		})
 	case HandlerGetContent:
-		contentRequest := &requests.Content{}
-		processIfJSONIsOk(json.Unmarshal(jsonBytes, &contentRequest), func() {
+		contentRequest := new(requests.Content)
+		processIfJSONIsOk(json.NewDecoder(rdr).Decode(contentRequest), func() {
 			reply, apiErr = r.GetContent(contentRequest)
 		})
 	case HandlerGetNodes:
-		nodesRequest := &requests.Nodes{}
-		processIfJSONIsOk(json.Unmarshal(jsonBytes, &nodesRequest), func() {
+		nodesRequest := new(requests.Nodes)
```

Comment:
Is there a benefit of using the new constructor compared to using the &..{} notation?

Reply:
A question no one can answer in the Go community 😎.
But I use it whenever there are no fields to initialize, otherwise I use alwaus `&Type{Field:X}`